datasource/src/main/scala/quasar/physical/s3/package.scala

```diff
@@ -45,4 +49,16 @@ object S3JsonParsing {
 
 package object s3 {
   val datasourceKind: DatasourceType = DatasourceType("s3", 1L)
+
+  @SuppressWarnings(Array("org.wartremover.warts.Recursion"))
+  def resourceCleanup[F[_]: FlatMap, A](r: Resource[F, A]): F[Unit] =
```

Comment:
Not that it's really a problem, but is there a reason to use `FlatMap` here rather than `scalaz.Bind`? We've mostly standardized on using scalaz types in signatures and then shims to convert between the two, which is why I ask. NBD either way.